### PR TITLE
Added TypeScript support

### PR DESCRIPTION
**Pull request by flokol120**
I lifted the whole project to TypeScript. I wanted you, @oja, to confirm this because this is quite a big thing. The commits are a bit messy, sorry for that. What I did is just adjusting the existing code to work with TS and adjusting the builder settings to get along with the TS files. The only file not affected by this is the injector file because otherwise it wouldn't work on Amazon Music's end.

### CODE REVIEW

File: const.ts [hunk 1]
```diff
@@ -4,5 +4,5 @@
  * @create date 2018-08-26 02:37:53
  * @modify date 2018-08-26 03:26:37
  * @desc this projects constants
-*/
-exports.APP_NAME = 'Amazon Music'
\ No newline at end of file
+ */
+export default "Amazon Music" as string;
```
Comment (BrunoScheufler) [inline const.ts]: Maybe keep the named export, rather than turning it into a default export. Just for future use

File: main/renderer.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+ * @author Flo Dörr <flo@dörr.site>
+ */
+ipcRenderer.on("nextTrack", () => {
+  webview.executeJavaScript("__am.nextTrack();", false, () => {
+    clearLyricsAndRestart();
+  });
+});
+
+/**
+ * handels "the previous track" key
```
Comment (BrunoScheufler) [inline main/renderer.ts]: `handels` > `handles` 😄

Comment (flokol120) [reply main/renderer.ts]: whoops :rofl:

File: main.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+ * handels the "play and pause" key
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+function playAndPause() {
+  mainWindow.webContents.send("playAndPause");
+}
+
+/**
+ * handels "the next track" key
```
Comment (BrunoScheufler) [inline main.ts]: The evil `handels` again 😄

File: main.ts [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+function nextTrack() {
+  mainWindow.webContents.send("nextTrack");
+}
+
+/**
+ * handels "the previous track" key
```
Comment (BrunoScheufler) [inline main.ts]: The same goes for here

File: main.ts [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+function previousTrack() {
+  mainWindow.webContents.send("previousTrack");
+}
+
+/**
+ * handels "the previous track" key
```
Comment (BrunoScheufler) [inline main.ts]: and here!

File: main/renderer.ts (hunk 2 above)
Comment (flokol120) [reply main/renderer.ts]: --> https://github.com/oja/amazon-music-linux/pull/15/commits/432bb855a1e44e6c29de0df4b6d98fb97e5c6b28

File: main.ts (hunk 3 above)
Comment (flokol120) [reply main.ts]: copy and paste is the evel :tongue: --> 432bb85

File: main.ts (hunk 4 above)
Comment (flokol120) [reply main.ts]: --> 432bb85

File: main.ts (hunk 5 above)
Comment (flokol120) [reply main.ts]: --> 432bb85

File: const.ts (hunk 1 above)
Comment (flokol120) [reply const.ts]: --> 7a811ff

Comment (flokol120) [reply const.ts]: --> 59d8c62

**Comment by oja** [comment]
@flokol120 all good! feel free to go ahead and merge!

**Pull request merged**